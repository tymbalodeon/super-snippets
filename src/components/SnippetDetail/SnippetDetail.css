.Snippet__detail__wrapper {
  margin-top: 50px;
}

.Snippet__info {
  color: #cbccc6;
}

.code {
  margin-bottom: 20px;
}

.button-detail {
  margin-left: 5px;
  margin-right: 5px;
}
