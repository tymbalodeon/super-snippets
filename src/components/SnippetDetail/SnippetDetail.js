import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SnippetApiService from '../../services/snippet-api-service';
import './SnippetDetail.css';

export default class SnippetDetail extends Component {
  state = {
    snippet: {}
  };

  handleDelete = () => {
    const idToDelete = this.props.match.params.snippet_id;
    SnippetApiService.deleteSnippet(idToDelete);
    this.props.history.goBack();
  };

  componentDidMount() {
    SnippetApiService.getSnippet(this.props.match.params.snippet_id).then(
      snippet => this.setState({ snippet })
    );
  }

  render() {
    return (
      <div className="Snippet__detail__wrapper">
        <div className="Snippet__details">
          <div className="Snippet__text">
            <h2 className="Snippet__heading">
              {this.state.snippet.snippet_name}
            </h2>
            <p className="Snippet__info">{this.state.snippet.info}</p>
            <div className="code">
              <code className="Snippet__content">
                {this.state.snippet.content}
              </code>
            </div>
          </div>
        </div>
        <div className="button-flex">
          <button className="button-detail">
            <Link to="#" onClick={this.props.history.goBack}>
              Back
            </Link>
          </button>
          <button className="button-detail">
            <Link
              to={{
                pathname: '/update',
                state: {
                  snippet: this.state.snippet
                }
              }}
            >
              Edit
            </Link>
          </button>
          <button className="button-detail-play">PLAY!</button>
          <button className="button-detail-delete" onClick={this.handleDelete}>
            Delete
          </button>
        </div>
      </div>
    );
  }
}
