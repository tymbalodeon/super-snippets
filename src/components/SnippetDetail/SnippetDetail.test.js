import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import SnippetDetail from './SnippetDetail';

describe('<SnippetDetail />', () => {
  const history = createMemoryHistory('/');
  const match = {
    params: {
      snippet_id: 1
    }
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <SnippetDetail history={history} match={match} />
      </BrowserRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SnippetDetail history={history} match={match} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
