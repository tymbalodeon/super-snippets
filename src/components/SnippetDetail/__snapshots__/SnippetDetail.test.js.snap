// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnippetDetail /> renders the UI as expected 1`] = `
<div
  className="Snippet__detail__wrapper"
>
  <div
    className="Snippet__details"
  >
    <div
      className="Snippet__text"
    >
      <h2
        className="Snippet__heading"
      />
      <p
        className="Snippet__info"
      />
      <div
        className="code"
      >
        <code
          className="Snippet__content"
        />
      </div>
    </div>
  </div>
  <div
    className="button-flex"
  >
    <button
      className="button-detail-back"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        BACK
      </a>
    </button>
    <button
      className="button-detail-edit"
    >
      <a
        href="/update"
        onClick={[Function]}
      >
        EDIT
      </a>
    </button>
    <button
      className="button-detail-play"
    >
      PLAY
    </button>
    <button
      className="button-detail-delete"
      onClick={[Function]}
    >
      DELETE
    </button>
  </div>
</div>
`;
